src/user/userService.ts

```diff
@@ -1,89 +0,0 @@
-import bcrypt from 'bcryptjs';
-import { BadRequest, UnprocessableEntity, Unauthorized } from 'http-errors';
-import { UserInput } from './interfaces';
-import { User } from '../libs/db/schema';
-import UserRepository from './userRepository';
-import TokenService from '../libs/common/tokenService';
-import EmailService from './emailService';
-
-class UserService {
-    constructor(
-        private userRepository = UserRepository,
-        private tokenService = TokenService,
-        private emailService = EmailService,
-    ) { }
-
-    async registerUser(input: UserInput): Promise<User> {
-        const { email, password } = input;
-
-        if (!email || !password) {
-            throw new BadRequest('Please provide email and password');
-        }
-
-        const userEmail = email.trim();
-        const userPassword = password.trim();
-
-        await this.checkUserExistence(userEmail);
-
-        const hashedPassword = await bcrypt.hash(userPassword, 10);
-
-        const verificationToken = this.tokenService.generateToken();
-
-        const newUser = await this.userRepository.create({
-            email: userEmail,
-            password: hashedPassword,
-            verification_token: verificationToken,
-        });
-
-        await this.emailService.sendVerificationEmail(userEmail, verificationToken);
-
-        return newUser;
```

Comment:
We should return the token as well

Reply:
@OmarMuhtaseb but right after registering the user still didnt verify his email and doesnt have access. so shouldn't we add the token on email verification and not signup. wdyt?

Reply:
@moaz-mostafa23 mmm not actually. Let's think of the flow here, if a user create an account, but didn't activate. Then, few days later, the user tries to login, it won't be possible, and also it won't be possible to create a new account, right?